### PR TITLE
Some Sonar cleanup

### PR DESCRIPTION
**Pull request by Sfiguz7**
## Description
Most of this is self explanatory, just some general cleanup. This includes adding @Overrides here and there, removing unused parameters/constructors (and then finding out I broke STB :unamused: ), reducing Cognitive Complexity by either moving code to a new method or removing unnecessary IFs.

## Changes
SonarCloud smells fixes, more info above

## Related Issues
N/A

## Checklist
- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java [hunk 1]
```diff
@@ -411,14 +411,15 @@ else if (dyeableSlot >= 0 && getPaintLevel() > 0) {
     }
 
     private DyeColor mixDyes(DyeColor dye1, DyeColor dye2) {
+        if (dye1 == dye2) {
+            return dye1;
+        }
+
         if (dye1.compareTo(dye2) > 0) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java:418]: You could make that an `else if`

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java [hunk 2]
```diff
@@ -449,9 +450,8 @@ else if (dye1 == DyeColor.PINK && dye2 == DyeColor.PURPLE) {
         else if (dye1 == DyeColor.WHITE && dye2 == DyeColor.GRAY) {
             return DyeColor.LIGHT_GRAY;
         }
-        else {
-            // colors don't mix
-            return null;
-        }
+
+        // colors don't mix
+        return null;
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java:455]: Why did you remove the `else` here?

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java (hunk 1 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java:418]: True, will do rn

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java (hunk 2 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java:455]: Makes Sonar happy :lul: All it does is reduce cognitive complexity by 1 bringing the total count to the 20 Sonar likes, it's nothing exceptionally useful really. I liked it more with the else as it seems clearer to me but this isn't that bad and reduces technical debt so I didn't see a good reason not to

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java:455]: :NotLikeThis:

Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java:455]: While I'm adding the else for the comment above, do you want me to revert this change or?

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/blocks/PaintCan.java:455]: yeah sure

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java [hunk 3]
```diff
@@ -60,14 +60,6 @@ public MultiBuilder() {
...
     }
 
-    public MultiBuilder(ConfigurationSection conf) {
-        super(conf);
-        mode = Mode.valueOf(conf.getString("mode"));
-        charge = conf.getDouble("charge");
-        String s = conf.getString("material");
-        material = s.isEmpty() ? null : Material.matchMaterial(s);
-    }
-
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:70]: Pretty sure that constructor is required...

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java (hunk 3 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:70]: IntelliJ marked it in grey and the project compiles fine after removing it, it's the only machine for which it's greyed out though. I guess I can add it back

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:70]: I am not 100% sure but I think that constructor is invoked via reflection, so it may not show as used even if it is.

Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:70]: Makes sense

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java [hunk 4]
```diff
@@ -532,21 +532,25 @@ public void run() {
...
-                                if ((x != 0 || y != 0 || z != 0) && b1.getType() == rec.source && STBUtil.isExposed(b1)) {
-                                    queue.offer(new SwapRecord(rec.player, b1, rec.source, rec.target, rec.layersLeft - 1, rec.builder, slot, rec.chargeNeeded));
-                                }
+                queueNextSet(rec, b, slot);
+
+                didWork = true;
+            }
+        }
+
+        private void queueNextSet(SwapRecord rec, Block b, int slot){
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:541]: Nonnull please :eyes:

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java [hunk 5]
```diff
@@ -532,21 +532,25 @@ public void run() {
...
+            if (rec.layersLeft > 0) {
+                for (int x = -1; x <= 1; x++) {
+                    for (int y = -1; y <= 1; y++) {
+                        for (int z = -1; z <= 1; z++) {
+                            Block b1 = b.getRelative(x, y, z);
+
+                            if ((x != 0 || y != 0 || z != 0) && b1.getType() == rec.source && STBUtil.isExposed(b1)) {
+                                queue.offer(new SwapRecord(rec.player, b1, rec.source, rec.target, rec.layersLeft - 1, rec.builder, slot, rec.chargeNeeded));
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:549]: Queue#offer() returns a boolean that determines whether the operation was successful or not.
This shouldn't be ignored, so I'd check that and return if no more items were able to be added.

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java (hunk 4 above)
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:541]: Space before the curly bracket please.

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java [hunk 6]
```diff
@@ -532,21 +532,25 @@ public void run() {
...
+                didWork = true;
+            }
+        }
+
+        private void queueNextSet(SwapRecord rec, Block b, int slot){
+            if (rec.layersLeft > 0) {
+                for (int x = -1; x <= 1; x++) {
+                    for (int y = -1; y <= 1; y++) {
+                        for (int z = -1; z <= 1; z++) {
+                            Block b1 = b.getRelative(x, y, z);
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:546]: bw is a bit of a dull name, how about "relBlock" or just "block"?

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java (hunk 5 above)
Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:549]: What do you mean "return if no more items were able to be added"? I can make the method a bool and return false if Queue#offer() returns false but then I'd have no real use for it in run() just above

Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:549]: I meant that you should probably do `return;` to break out of the loop if the queue cannot accept any more items, since we can basically stop at that point.

Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:549]: Aaah lol yea that would be wise

Comment (Sfiguz7) [reply src/main/java/io/github/thebusybiscuit/sensibletoolbox/items/MultiBuilder.java:549]: It makes for a long ass if clause, any idea how to make it prettier?

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM

**Review by poma123 — APPROVED** [review]
